### ISSUE TITLE
ReLU node causes numeric instability

### ISSUE THREAD

**Issue by emjotde** [emjotde/Marian#14]
When running the MNIST example with ReLU for more than 25 epochs NAN appears. This seems to be a known issue with ReLU and crossentropy, and the suggested solution is some sort of clipping. This requires more investigation.

**Comment by emjotde** [comment]
Replacing softmax with log-softmax in the cross-entropy layer fixed this.
